Documentation/man/man3/ConfigFile.3
.TH "ConfigFile" 3 "Version 0.0.0.2" "My Project" \" -*- nroff -*-
.ad l
.nh
.SH NAME
ConfigFile
.SH SYNOPSIS
.br
.PP
.PP
\fR#include <D_CONFIGFILE\&.h>\fP
.PP
Inherited by \fBAiObject\fP, \fBBuidling\fP, \fBComputeShader\fP, \fBFragmentShader\fP, \fBGeometryShader\fP, \fBGlobal\fP, \fBIsland\fP, \fBManager\fP, \fBMaterial\fP, \fBParticle\fP, \fBProjectile\fP, \fBVehicle\fP, and \fBVertexShader\fP\&.
.SS "Public Types"

.in +1c
.ti -1c
.RI "enum \fBConfigType\fP { \fBBuilding\fP, \fBVehicle\fP, \fBProjectile\fP, \fBParticle\fP, \fBIsland\fP, \fBAiObject\fP, \fBGlobal\fP, \fBManager\fP, \fBVertexShader\fP, \fBFragmentShader\fP, \fBGeometryShader\fP, \fBComputeShader\fP, \fBMaterial\fP }"
.br
.in -1c
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBConfigFile\fP (std::string name, std::string modelpath, std::vector< std::string > texturepath, \fBConfigType\fP configType, nlohmann::json data)"
.br
.ti -1c
.RI "virtual void \fBPalceholder\fP ()"
.br
.in -1c
.SS "Static Public Member Functions"

.in +1c
.ti -1c
.RI "static std::string \fBtoString\fP (\fBConfigType\fP type)"
.br
.ti -1c
.RI "static \fBConfigType\fP \fBtoConfigType\fP (const std::string &str)"
.br
.in -1c
.SS "Public Attributes"

.in +1c
.ti -1c
.RI "std::string \fBName\fP"
.br
.ti -1c
.RI "std::string \fBModelpath\fP"
.br
.ti -1c
.RI "std::vector< std::string > \fBTexturepaths\fP"
.br
.ti -1c
.RI "\fBConfigType\fP \fBConfigType_\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line \fB12\fP of file \fBD_CONFIGFILE\&.h\fP\&.
.SH "Member Enumeration Documentation"
.PP 
.SS "enum \fBConfigFile::ConfigType\fP"

.PP
\fBEnumerator\fP
.in +1c
.TP
\fB\fIBuilding \fP\fP
.TP
\fB\fIVehicle \fP\fP
.TP
\fB\fIProjectile \fP\fP
.TP
\fB\fIParticle \fP\fP
.TP
\fB\fIIsland \fP\fP
.TP
\fB\fIAiObject \fP\fP
.TP
\fB\fIGlobal \fP\fP
.TP
\fB\fIManager \fP\fP
.TP
\fB\fIVertexShader \fP\fP
.TP
\fB\fIFragmentShader \fP\fP
.TP
\fB\fIGeometryShader \fP\fP
.TP
\fB\fIComputeShader \fP\fP
.TP
\fB\fIMaterial \fP\fP
.PP
Definition at line \fB18\fP of file \fBD_CONFIGFILE\&.h\fP\&.
.nf
18                     {
19         Building,
20         Vehicle,
21         Projectile,
22         Particle,
23         Island,
24         AiObject,
25         Global,
26         Manager,
27         VertexShader,
28         FragmentShader,
29         GeometryShader,
30         ComputeShader,
31         Material
32     };
.PP
.fi

.SH "Constructor & Destructor Documentation"
.PP 
.SS "ConfigFile::ConfigFile (std::string name, std::string modelpath, std::vector< std::string > texturepath, \fBConfigType\fP configType, nlohmann::json data)"

.PP
Definition at line \fB5\fP of file \fBD_CONFIGFILE\&.cpp\fP\&.
.nf
6 {
7     Name = name;
8     Modelpath = modelpath;
9     Texturepaths = texturepath;
10     ConfigType_ = configType;
11     //Data = data;
12 }
.PP
.fi

.PP
References \fBConfigType_\fP, \fBModelpath\fP, \fBName\fP, and \fBTexturepaths\fP\&.
.SH "Member Function Documentation"
.PP 
.SS "virtual void ConfigFile::Palceholder ()\fR [inline]\fP, \fR [virtual]\fP"

.PP
Definition at line \fB74\fP of file \fBD_CONFIGFILE\&.h\fP\&.
.nf
74 {};
.PP
.fi

.SS "static \fBConfigType\fP ConfigFile::toConfigType (const std::string & str)\fR [inline]\fP, \fR [static]\fP"

.PP
Definition at line \fB52\fP of file \fBD_CONFIGFILE\&.h\fP\&.
.nf
52                                                          {
53         if (str == "Building") return Building;
54         if (str == "Vehicle") return Vehicle;
55         if (str == "Projectile") return Projectile;
56         if (str == "Particle") return Particle;
57         if (str == "Island") return Island;
58         if (str == "AiObject") return AiObject;
59         if (str == "Global") return Global;
60         if (str == "Manager") return Manager;
61         if (str == "VertexShader")return VertexShader;
62         if (str == "FragmentShader")return FragmentShader;
63         if (str == "GeometryShader")return GeometryShader;
64         if (str == "ComputeShader")return ComputeShader;
65         if (str == "Material") return Material;
66         throw std::invalid_argument("Unknown ConfigType: " + str);
67     }
.PP
.fi

.PP
References \fBAiObject\fP, \fBBuilding\fP, \fBComputeShader\fP, \fBFragmentShader\fP, \fBGeometryShader\fP, \fBGlobal\fP, \fBIsland\fP, \fBManager\fP, \fBMaterial\fP, \fBParticle\fP, \fBProjectile\fP, \fBVehicle\fP, and \fBVertexShader\fP\&.
.PP
Referenced by \fBCORE::ConfigLoader::LoadCOnfigFile()\fP\&.
.SS "static std::string ConfigFile::toString (\fBConfigType\fP type)\fR [inline]\fP, \fR [static]\fP"

.PP
Definition at line \fB33\fP of file \fBD_CONFIGFILE\&.h\fP\&.
.nf
33                                                {
34         switch (type) {
35         case Building: return "Building";
36         case Vehicle: return "Vehicle";
37         case Projectile: return "Projectile";
38         case Particle: return "Particle";
39         case Island: return "Island";
40         case AiObject: return "AiObject";
41         case Global: return "Global";
42         case Manager: return "Manager";
43         case VertexShader: return "VertexShader";
44         case FragmentShader: return "FragmentShader";
45         case GeometryShader: return "GeometryShader";
46         case ComputeShader: return "ComputeShader";
47         case Material: return "Material";
48         default: return "Unknown";
49         }
50     }
.PP
.fi

.PP
References \fBAiObject\fP, \fBBuilding\fP, \fBComputeShader\fP, \fBFragmentShader\fP, \fBGeometryShader\fP, \fBGlobal\fP, \fBIsland\fP, \fBManager\fP, \fBMaterial\fP, \fBParticle\fP, \fBProjectile\fP, \fBVehicle\fP, and \fBVertexShader\fP\&.
.SH "Member Data Documentation"
.PP 
.SS "\fBConfigType\fP ConfigFile::ConfigType_"

.PP
Definition at line \fB72\fP of file \fBD_CONFIGFILE\&.h\fP\&.
.PP
Referenced by \fBConfigFile()\fP\&.
.SS "std::string ConfigFile::Modelpath"

.PP
Definition at line \fB70\fP of file \fBD_CONFIGFILE\&.h\fP\&.
.PP
Referenced by \fBConfigFile()\fP\&.
.SS "std::string ConfigFile::Name"

.PP
Definition at line \fB69\fP of file \fBD_CONFIGFILE\&.h\fP\&.
.PP
Referenced by \fBConfigFile()\fP\&.
.SS "std::vector<std::string> ConfigFile::Texturepaths"

.PP
Definition at line \fB71\fP of file \fBD_CONFIGFILE\&.h\fP\&.
.PP
Referenced by \fBConfigFile()\fP\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for My Project from the source code\&.


UPRISE_ENGINE/CONFIGFILE.cpp
#include "pch.h"

#include "CONFIGFILE.h"

ConfigFile::ConfigFile(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data)
{
	Name = name;
	Modelpath = modelpath;
	Texturepaths = texturepath;
	ConfigType_ = configType;
	//Data = data;
}



  Buidling::Buidling(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data) 
	  : ConfigFile(name, modelpath, texturepath, configType, data) {
	 // buildinngType = data["BuildingType"];
	  switch (buildinngType) {
	  case Residential:
	  {
		  std::cout << "Building Type: Residential" << std::endl;
		  break;
	  }
	  case Needs:
	  {
		  std::cout << "Building Type: Needs" << std::endl;
		  break;
	  }
	  case Production_Raw:
	  {
		  std::cout << "Building Type: Production Raw Materials" << std::endl;
		  break;
	  }
	  case Production_Not_Raw:
	  {
		  std::cout << "Building Type: Production Non-Raw Materials" << std::endl;
		  break;
	  }
	  case Energy_In:
	  {
		  std::cout << "Building Type: Energy Industrial" << std::endl;
		  break;
	  }
	  case Energy_I_W:
	  {
		  std::cout << "Building Type: Energy Industrial with Waste" << std::endl;
		  break;
	  }
	  case Energy_R:
	  {
		  std::cout << "Building Type: Renewable Energy" << std::endl;
		  break;
	  }
	  case Energy_R_W:
	  {
		  std::cout << "Building Type: Renewable Energy with Waste" << std::endl;
		  break;
	  }
	  case Energy:
	  {
		  std::cout << "Building Type: General Energy" << std::endl;
		  break;
	  }
	  case Eco:
	  {
		  std::cout << "Building Type: Ecological" << std::endl;
		  break;
	  }
	  case Decoration:
		  std::cout << "Building Type: Decoration" << std::endl;
		  break;
	  case Military:
		  std::cout << "Building Type: Military" << std::endl;
		  break;
	  case Special:
		  std::cout << "Building Type: Special" << std::endl;
		  break;
	  default:
		  std::cout << "Unknown Building Type" << std::endl;
		  break;
	  }
  
  }

  Vehicle::Vehicle(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data)
	  : ConfigFile(name, modelpath, texturepath, configType, data) {
  
  }

  Projectile::Projectile(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data)
	  : ConfigFile(name, modelpath, texturepath, configType, data) {
  
  }

  Particle::Particle(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data)
	  : ConfigFile(name, modelpath, texturepath, configType, data) {
  
  }

  Island::Island(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data)
	  : ConfigFile(name, modelpath, texturepath, configType, data) {
  
  }

  AiObject::AiObject(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data) 
	  : ConfigFile(name, modelpath, texturepath, configType, data) {
  
  }

  Global::Global(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data) 
	  : ConfigFile(name, modelpath, texturepath, configType, data) {
  
  }

  Manager::Manager(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data) 
	  : ConfigFile(name, modelpath, texturepath, configType, data) {
 
  }

  FragmentShader::FragmentShader(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data) 
	  : ConfigFile(name, modelpath, texturepath, configType, data) {
	  string shadercode="";
	  for (string s : data["shaderCode"]) {
		  shadercode = shadercode + s+"\n";
	  }
	  ShaderCode = shadercode;
  
  }

  GeometryShader::GeometryShader(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data)
	  : ConfigFile(name, modelpath, texturepath, configType, data) {
	  string shadercode = "";
	  for (string s : data["shaderCode"]) {
		  shadercode = shadercode + s;
	  }
	  ShaderCode = shadercode;
  }

  Material::Material(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data) 
	  : ConfigFile(name, modelpath, texturepath, configType, data) {
	  string s = data["ShaderCount"];
	  ShaderCount = s.data()[0]-48;
	  VertexShader = data["VertexShader"];
	  FragmentShader = data["FragmentShader"];

	 // GeometryShader = Data[3];
  }

  VertexShader::VertexShader(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data) 
	  : ConfigFile(name, modelpath, texturepath, configType, data) {
	  string shadercode = "";
	  for (string s : data["shaderCode"]) {
		  shadercode = shadercode + s+"\n";
	  }
	  ShaderCode = shadercode;

  }

  ComputeShader::ComputeShader(std::string name, std::string modelpath, std::vector<std::string> texturepath, ConfigType configType, nlohmann::json data)
	  : ConfigFile(name, modelpath, texturepath, configType, data) {
	  string shadercode = "";
	  for (string s : data["shaderCode"]) {
		  shadercode = shadercode + s;
	  }
	  ShaderCode = shadercode;

  }
